force-app/main/default/classes/Gifts.cls

```diff
@@ -146,7 +180,8 @@ public with sharing class Gifts {
     private void setElevateBatchIds(List<DataImport__c> dataImports) {
         elevateBatchIds = new Set<String>();
         for (DataImport__c dataImport : dataImportsById.values()) {
-            if (dataImport?.Payment_Status__c == ElevateTransactionStatus.AUTHORIZED.name()) {
+            if (dataImport?.Status__c == BDI_DataImport_API.bdiDryRunValidated
```

Comment:
Non-blocking: Can we extract this guard clause into a method of its own so we have an understandable reference point always?